### ISSUE TITLE
tracker for ostree native container conversion

### ISSUE THREAD

**Issue by cgwalters** [fedora-silverblue/issue-tracker#359]
Hi, filing this in regards to https://fedoraproject.org/wiki/Changes/OstreeNativeContainer and
https://pagure.io/releng/issue/11047
specifically.

Any objections to taking this first step of generating `quay.io/fedora/fedora-silverblue:37` etc.?

**Comment by travier** [comment]
+1 from me. I've started work to setup CI with a similar idea in https://gitlab.com/fedora/ostree/ci-test.

Maybe we need to make a new change request or merge than change into the unified core change request to make that happen for 38.

**Comment by travier** [comment]
This is related to https://github.com/fedora-silverblue/issue-tracker/issues/334.

I have a working pipeline run in https://gitlab.com/fedora/ostree/ci-test/-/pipelines which uses rpm-ostree compose-image and pushed the result to https://quay.io/repository/fedora-ostree-desktops/silverblue?tab=tags & https://quay.io/repository/fedora-ostree-desktops/kinoite?tab=tags.

Next step is to figure out why lorax does not successfully create the installation media: https://gitlab.com/fedora/ostree/ci-test/-/issues/1

And then I need to merge ideas from https://gitlab.com/fedora/sigs/sway/ostree-config/-/blob/f37-sway-spin/.gitlab-ci.yml & https://gitlab.com/fedora/sigs/sway/fedora-kickstarts/-/blob/main-sway/.gitlab-ci.yml and look at enabling installation tests with a kickstart to get bootable QEMU images from the CI.

**Comment by cgwalters** [comment]
> I have a working pipeline run in https://gitlab.com/fedora/ostree/ci-test/-/pipelines which uses rpm-ostree compose-image

Nice, thanks for pushing this forward!  You should use `rpm-ostree compose image --format registry` to directly push to the registry - this crucially enables change detection.

**Comment by dustymabe** [comment]
bikeshed:

Why not just `quay.io/fedora/fedora-silverblue` and `quay.io/fedora/fedora-kinoite` to match what we're doing with `quay.io/fedora/fedora-coreos`? Are these just intermediate (testing) locations?

**Comment by travier** [comment]
> Why not just `quay.io/fedora/fedora-silverblue` and `quay.io/fedora/fedora-kinoite` to match what we're doing with `quay.io/fedora/fedora-coreos`? Are these just intermediate (testing) locations?

Yes, those are testing locations and testing repo. I'll request access to the official namespace once we have something more complete, and we will likely want Pungi to do the builds and pushes, not this CI here.

**Comment by travier** [comment]
> Yes, those are testing locations and testing repo. I'll request access to the official namespace once we have something more complete, and we will likely want Pungi to do the builds and pushes, not this CI here.

To add to that, the general idea behind https://github.com/fedora-silverblue/issue-tracker/issues/334 is to be able to make a PR and then get an image and later also an installer ISO and have tests run for it automatically, similarly to what we have for FCOS.

I don't know how exactly we'll make the official builds in the Fedora infra but I think the builds pushed by the GitLab CI will likely stay "non-official".

**Comment by travier** [comment]
Folks are starting to do cool things with this!

- https://discussion.fedoraproject.org/t/booting-silverblue-kinoite-using-podman-docker/44457/7
- https://discussion.fedoraproject.org/t/feature-for-custom-rpm-ostree-container-native-builds/44480

**Comment by travier** [comment]
We now have official images for all arches in Rawhide!

https://kojipkgs.fedoraproject.org/compose/rawhide/latest-Fedora-Rawhide/compose/Silverblue/x86_64/images/

Now we'll have to upload them to the registry.

**Comment by travier** [comment]
I'll close this issue in favor of the one in the Fedora Atomic Desktop issue tracker: https://gitlab.com/fedora/ostree/sig/-/issues/2
